tools/bowtie2.wdl

```diff
@@ -0,0 +1,410 @@
+
+    parameter_meta {
+        reference: "Reference genome in FASTA format (optionally gzipped)"
+        prefix: "Prefix for the Bowtie2 index files"
+        ncpu: "Number of threads to use"
+    }
+
+    input {
+        File reference
+        String prefix
+        Int ncpu = 4
+    }
+
+    String base = basename(reference, ".gz")
+
+    command <<<
+        set -euo pipefail
+
+        gunzip -c ~{reference} > ~{base} \
+           || ln -sf ~{reference} ~{base}
+
+        bowtie2-build --threads=~{ncpu} ~{base} ~{prefix}
+
+        tar -czf ~{prefix}.tar.gz ~{prefix}.*
+    >>>
+
+    output {
+        Array[File] index_files = glob("*.bt2")
+        File bowtie_db_tar_gz = prefix + ".tar.gz"
+    }
+
+    runtime {
+        cpu: ncpu
+        memory: "20 GB"
+        container: "ghcr.io/stjudecloud/bowtie2:2.5.4"
+        maxRetries: 1
+    }
+}
+
+task align {
+    meta {
+        description: "Aligns reads to a reference genome using Bowtie2"
+        outputs: {
+            aligned_bam: "Aligned reads in BAM format",
+            unpaired_unaligned: "Unpaired reads that didn't align",
+            unpaired_aligned: "Unpaired reads that aligned at least once",
+            paired_discordant: "Pairs that didn't align concordantly",
+            paired_concordant: "Pairs that aligned concordantly at least once",
+            metrics: "Metrics file",
+        }
+    }
+
+    parameter_meta {
+        bowtie_db_tar_gz: "Bowtie2 index files in tarred, gzipped format"
+        read_one_fastq_gz: "A gzipped FASTQ file containing read one information"
+        read_two_fastq_gz: "A gzipped FASTQ file containing read two information"
+        skip: "Skip the first N reads/read pairs"
+        upto: "Only align the first N reads/read pairs"
+        trim5: "Trim N bases from the 5' end of each read before alignment"
+        trim3: "Trim N bases from the 3' end of each read before alignment"
+        seed_mismatch: {
+            description: "Maximum number of mismatches in the seed [0, 1] during multiseed alignment.",
+            external_help: "https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic",
+            bowtie_option: "-N",
+        }
+        seed_substring: {
+            description: "Length of seed substring. >3, <32",
+            external_help: "https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic",
+            bowtie_option: "-L",
+        }
+        dpad: "Include N extra reference characters on sides of dynamic programming table"
+        gbar: "Disallow gaps within N nucleotides of read ends"
+        ignore_quals: "When calculating a mismatch penalty, always consider the quality value at the mismatched position to be the highest possible, regardless of the actual value. Treat all quality values as 30 on Phred scale."
+        nofw: "Do not align to the forward (original) strand of the reference"
+        norc: "Do not align to the reverse-complement strand of the reference"
```

Comment:
Again, wondering the use case here. This seems odd to me.

Reply:
I implemented `bowtie2` under the assumption that we will move forward with the separate `tools` as a resource repo model. For that, I think having the complete set of parameters exposed is preferable.

Reply:
relevant comment I've made: https://github.com/stjudecloud/workflows/issues/168#issuecomment-2297434024

IMO sometimes tool authors provide too much parameterization and some options should not be exposed. This strikes me as one such case. However silently omitting options is potentially confusing to users (or seems like a mistake on our part), so I suggest we document any parameters we opt to exclude somewhere in the meta. Perhaps under `meta.omitted_parameters`? I'm not sure how much detail or rationale to include...